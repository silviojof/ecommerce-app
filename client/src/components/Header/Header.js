import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from "react-router";
import Logo_ML from '../../images/Logo_ML.png';
import Logo_ML2x from '../../images/Logo_ML@2x.png';
import ic_Search from '../../images/ic_Search.png';
import ic_Search2x from '../../images/ic_Search@2x.png';

import styles from './Header.module.scss';

export const Header = ({ history }) => {
    const [search, setSearch] = useState('');
    const submit = (e) => {
        e.preventDefault();
        history.push(`/items?search=${search}`)
    }
    return (
        <header className={styles['nav-header']}>
            <div className={styles.container}>
                <a href="/" className={styles.logo}>
                    <img src={Logo_ML} srcSet={`${Logo_ML2x} 2x`} alt="logo" className={styles.image} />
                </a>
                <form onSubmit={submit} className={styles.form}>
                    <div className={styles['form-container']}>
                        <input
                            className={styles.input}
                            aria-label="Search Item"
                            type="text"
                            placeholder="Nunca dejes de buscar"
                            onChange={e => setSearch(e.target.value)}
                            value={search}
                        />
                        <button type="submit" data-testid="submit-button" className={styles.button}>
                            <img src={ic_Search} alt="bar" srcSet={`${ic_Search2x} 2x`} />
                        </button>
                    </div>
                </form>
            </div>
        </header>
    )
};

Header.propTypes = {
    history: PropTypes.object.isRequired,
};

export default withRouter(Header);