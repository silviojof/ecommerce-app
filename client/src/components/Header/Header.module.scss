@import '../../styles/colors';

.nav-header {
    background-color: $yellow;
    border: 0;
    position: relative;
}

.logo {
    text-decoration: none;
    color: inherit;
    outline: 0;
    cursor: pointer;
}

.form-container {
    display: flex;
    flex: 1;
    flex-direction: row;
    border-radius: $radius;
    height: 40px;
}

.form {
    flex: 1;
}

.image {
    width: 56px;
    margin-right: 24px;
}

.input {
    flex: 1;
    border: none;
    font-size: 18px;
    color: $mediumGrey;
    padding: 8px 16px;
    border-top-left-radius: $radius;
    border-bottom-left-radius: $radius;
    margin: 0;
    &:focus {
        outline: none;
    }
    &::placeholder {
        color: $mediumGrey;
    }
}
.button {
    background-color: $lightGrey;
    border: none;
    width: 40px;
    margin: 0;
    border-top-right-radius: $radius;
    border-bottom-right-radius: $radius;
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
}

.container {
    @extend %container;
    display: flex;
    align-items: center;
    padding: 10px 0;
}