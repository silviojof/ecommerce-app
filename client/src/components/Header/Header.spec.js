import React from 'react'
import {render, fireEvent,} from 'react-testing-library'
import { Header } from './Header'

const history = {
    push: jest.fn(),
};

it ('matches snapshot', () => {
    const { container } = render(<Header history={history} />);
    expect(container.firstChild).toMatchSnapshot();
});

it('changes on Type', () => {
    const {getByPlaceholderText} = render(<Header history={history} />);
    const input = getByPlaceholderText('Nunca dejes de buscar');
    fireEvent.change(input, { target: { value: 'search' } });
    expect(getByPlaceholderText('Nunca dejes de buscar').value).toEqual('search');
});

it('submits properlly', () => {
    const spy = jest.fn();
    const history = {
        push: spy,
    };
    const {getByTestId, getByPlaceholderText} = render(<Header history={history} />);
    const button = getByTestId('submit-button');
    const input = getByPlaceholderText('Nunca dejes de buscar');
    fireEvent.change(input, { target: { value: 'search' } });
    fireEvent.click(button);
    expect(spy).toHaveBeenCalledWith('/items?search=search');
});