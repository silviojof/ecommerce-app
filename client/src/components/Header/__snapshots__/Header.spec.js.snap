// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  class="nav-header"
>
  <div
    class="container"
  >
    <a
      class="logo"
      href="/"
    >
      <img
        alt="logo"
        class="image"
        src="Logo_ML.png"
        srcset="Logo_ML@2x.png 2x"
      />
    </a>
    <form
      class="form"
    >
      <div
        class="form-container"
      >
        <input
          aria-label="Search Item"
          class="input"
          placeholder="Nunca dejes de buscar"
          type="text"
          value=""
        />
        <button
          class="button"
          data-testid="submit-button"
          type="submit"
        >
          <img
            alt="bar"
            src="ic_Search.png"
            srcset="ic_Search@2x.png 2x"
          />
        </button>
      </div>
    </form>
  </div>
</header>
`;
